solidity/random-beacon/test/RandomBeacon.Relay.test.ts

```diff
@@ -257,78 +403,332 @@ describe("RandomBeacon - Relay", () => {
+            )
         ).to.be.revertedWith("No relay request in progress")
       })
     })
   })
 
+  describe("reportRelayEntryTimeout", () => {
+    beforeEach(async () => {
+      await createGroup(randomBeacon, signers)
+
+      await approveTestToken()
+      await randomBeacon.connect(requester).requestRelayEntry(ZERO_ADDRESS)
+    })
+
+    context("when relay entry timed out", () => {
+      context(
+        "when other active groups exist after timeout is reported",
+        () => {
+          let tx: ContractTransaction
+
+          beforeEach(async () => {
+            // Create another group in a rough way just to have an active group
+            // once the one handling the timed out request gets terminated.
+            // This makes the request retry possible. That group will not
+            // perform any signing so their public key can be arbitrary bytes.
+            // Also, that group is created just after the relay request is
+            // made to ensure it is not selected for signing the original request.
+            await (randomBeacon as RandomBeaconStub).roughlyAddGroup(
+              "0x01",
+              signersIDs
+            )
+
+            // `groupSize * relayEntrySubmissionEligibilityDelay +
+            // relayEntryHardTimeout`.
+            await mineBlocks(64 * 10 + 5760)
+
+            tx = await randomBeacon.reportRelayEntryTimeout()
+          })
+
+          it("should slash entire stakes of all group members", async () => {
+            await expect(tx)
+              .to.emit(staking, "Slashed")
+              .withArgs(to1e18(1000), signersAddresses)
+          })
+
+          it("should terminate the group", async () => {
+            // TODO: Implementation once `Groups` library is ready.
+          })
+
+          it("should emit RelayEntryTimedOut event", async () => {
+            await expect(tx)
+              .to.emit(randomBeacon, "RelayEntryTimedOut")
+              .withArgs(1)
+          })
+
+          it("should retry current relay request", async () => {
+            // We expect the same request ID because this is a retry.
+            // Group ID is still `0` because there is only one group
+            // after termination was performed.
+            await expect(tx)
+              .to.emit(randomBeacon, "RelayEntryRequested")
+              .withArgs(1, 0, blsData.previousEntry)
+
+            expect(await randomBeacon.isRelayRequestInProgress()).to.be.true
+          })
+        }
+      )
+
+      context("when no active groups exist after timeout is reported", () => {
+        let tx: ContractTransaction
+
+        beforeEach(async () => {
+          // `groupSize * relayEntrySubmissionEligibilityDelay +
+          // relayEntryHardTimeout`.
+          await mineBlocks(64 * 10 + 5760)
+
+          tx = await randomBeacon.reportRelayEntryTimeout()
+        })
+
+        it("should slash entire stakes of all group members", async () => {
```

Comment:
Same comment about the test name 😅

Reply:
Addressed in https://github.com/keep-network/keep-core/pull/2671/commits/e6ac2c4008995b5fb800737735d03c55061c3d03 being part of the follow-up PR: https://github.com/keep-network/keep-core/pull/2671